intro/README.md

```diff
@@ -0,0 +1,629 @@
+         a[0 .. 3] = b              // error, operands have different length
+    }
+    ```
+        
+    **.length** is a builtin array property. For an extensive list of array properties click [here](https://dlang.org/spec/arrays.html#array-properties).
+    
+2. Array Concatenation:
+
+    ```d
+    void main()
+    {
+        int[] a;
+        int[] b = [1, 2, 3];
+        int[] c = [4, 5];
+     
+        a = b ~ c;       // a will be [1, 2, 3, 4, 5]
+        a = b;           // a refers to b
+        a = b ~ c[0..0]; // a refers to a copy of b
+        a ~= c;          // equivalent to a = a ~ c;
+    }
+    ```
+       
+    Concatenation always creates a copy of its operands, even if one of the operands is a 0-length array. The operator `~=` does not always create a copy. For more information click [here](https://dlang.org/spec/arrays.html#resize).
+    
+3. Vector operations:
+
+    ```d
+    void main()
+    {
+        int[] a = [1, 2, 3];
+        int[] b;
+        int[] c;
+     
+        b[] = a[] + 4;         // b = [5, 6, 7]
+        c[] = (a[] + 1) * b[]  // c = [10, 18, 28]
+    }
+    ```
+    
+    Many array operations, also known as vector operations, can be expressed at a high level rather than as a loop. A vector operation is indicated by the slice operator appearing as the left-hand side of an assignment or an op-assignment expression. The right-hand side can be an expression consisting either of an array slice of the same length and type as the left-hand side or a scalar expression of the element type of the left-hand side, in any combination.
+
+    Using the concepts of slicing and concatenation, we can modify the original example (that does the removal of an element from the array) so that the **for** loop is no longer necessary:
+    
+    ```d
+    int main()
+    {
+    int position = 7, c, n = 10; 
+    int[] array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+ 
+    array = array[0 .. position] ~ array[position + 1 .. $]; 
+ 
+    import std.stdio;
+    writeln("Resultant array:");
+    writeln(array);                                                                                                                           
+ 
+    return 0;
+    }
+    ```
+        
+    [writeln](https://dlang.org/library/std/stdio/writeln.html) is a function from the standard D library that does not require a format and is easily usable with a plethora of types.
+
+    As you can see, the resulting code is much more expressive and fewer lines of code were utilized.
+
+### Quiz
+What is the output of the following code?
+
+```d
+void setLength(int[] a, size_t length)
+{
+    a.length = length;
+}
+
+void startWith2(int[] a)
+{
+    a[0] = 2;
+    a[1] = 2;
+}
+
+void main()
+{
+    import std.stdio : writeln;
+    int[] a = new int[5];
+    
+    a.setLength(2);
+    writeln(a.length);
+
+    a.startWith2();
+    writeln(a);
+}
+```
+    
+- [2, 2]
+- [2, 2, 0, 0, 0]
+- [0, 0, 0, 0, 0]
+- [0, 0]
```

Comment:
At this quiz you are also outputting the length, however the answers contain only the array.